### ISSUE TITLE
Support jekyll's -s command line switch

### ISSUE THREAD

**Issue by chepec** [inukshuk/jekyll-scholar#53]
Installed jekyll-scholar (v 4.0.4) on Debian 7 using `gem install jekyll-scholar`.
Used the default settings, created a barebones post, and building the site gave this error:
```
Liquid Exception: No such file or directory @ rb_sysopen - ./_bibliography/references.bib.bib in _posts/2014-07-18-jekyll-scholar.md
``` 

Here are the details. The content of the markdown file:
``` 
---
layout: post
title: "Adding bibliographies and citations"
date: 2014-07-18 16:00
tags: jekyll scholar
---

{{ page.title }}
================
Duis 'aute irure dolor in reprehenderit in voluptate' velit esse
cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat
cupidatat non proident, 'sunt in culpa qui officia deserunt 
mollit anim id est laborum'.

References
----------
{% bibliography %}
``` 

In `_plugins/ext.rb` I put the line `require 'jekyll/scholar'`.
I used the default scholar params in `_config.yml`:
```
scholar:
  style: apa
  locale: en
  sort_by: none
  order: ascending
  source: ./_bibliography
  bibliography: references.bib
  bibliography_template: "{{reference}}"
  replace_strings: true
  join_strings:    true
  details_dir:    bibliography
  details_layout: bibtex.html
  details_link:   Details
  query: "@*"
``` 
(Sidenote: all keys from ` bibliography_template` and down could be removed, jekyll build still gave the same error message.)

I changed the `'bibliography'` setting to just `references` (without the *.bib) but got the same error except it now pointed to the correct file:
```
Liquid Exception: No such file or directory @ rb_sysopen - ./_bibliography/references.bib in _posts/2014-07-18-jekyll-scholar.md
```

The bib-file was named `references.bib` and located in `_bibliography`, in the jekyll site source directory (next to `_posts`, `_plugins`, etc.). For good measure, here it is:
``` 
@book{Adachi1999,
author = {Adachi, Sadao},
editor = {Adachi, Sadao},
isbn = {978-0-7923-8567-7},
pages = {714},
publisher = {Springer Verlag},
title = {{Optical Constants of Crystalline and Amorphous Semiconductors - Numerical Data and Graphical Information}},
year = {1999}
}
@book{Albery1975,
address = {Oxford},
author = {Albery, John},
publisher = {Clarendon Press},
title = {{Electrode kinetics}},
year = {1975}
}
@book{Atkins2001,
author = {Atkins, Peter},
edition = {third},
isbn = {0-19-879290-5},
publisher = {Oxford University Press},
title = {{The Elements of Physical Chemistry}},
year = {2001}
}
@book{Atkins2002,
author = {Atkins, Peter and de Paula, Julio},
edition = {seventh},
isbn = {0-19-879285-9},
publisher = {Oxford University Press},
title = {{Atkins' Physical Chemistry}},
year = {2002}
}
@book{Attwood1999,
author = {Attwood, David},
isbn = {0-521-65214-6},
publisher = {Cambridge University Press},
title = {{Soft X-rays and extreme ultraviolet radiation Principles and Applications}},
year = {1999}
}
@phdthesis{Avendano2004,
author = {Avenda\~{n}o, Esteban},
school = {Uppsala university},
title = {{Electrochromism in Nickel-based Oxides}},
type = {PhD thesis},
url = {http://urn.kb.se/resolve?urn=urn:nbn:se:uu:diva-4307},
year = {2004}
}
@book{Bagotsky2006,
author = {Bagotsky, V. S.},
edition = {2},
isbn = {978-0-471-70058-6},
publisher = {Wiley-Interscience},
title = {{Fundamentals of electrochemistry}},
year = {2006}
}
``` 
And I checked, the bibtex file is UTF8/ASCII.

This error causes the site to completely fail to render, and I have found no way around it other than disabling jekyll-scholar and removing all its commands from the markdown file.
As far as I can tell, jekyll-scholar is actually installed properly in the gems folder. I am new to jekyll and ruby, so it could be that I am missing something obvious. Any suggestions?

**Comment by chepec** [comment]
Sorry to comment on a closed issue, but doesn't the error message `Liquid Exception: No such file or directory @ rb_sysopen - ./_bibliography/references.bib.bib in _posts/2014-07-07-rmarkdown-pandoc-knitr.md` indicate that the site is looking for `_bibliography/references.bib` in the `_posts` folder?
Should I perhaps specify `source: ./_bibliography` in `_config.yml` using some absolute path instead?

**Comment by inukshuk** [comment]
You're right, maybe Jekyll now changes directories when creating posts? (You run `jekyll build` when you are in the root folder, right?)

What you can try to do is to put the full path to your file into the `bibliography` option -- if that is an absolute path, `source` will be ignored.

**Comment by chepec** [comment]
I always run `jekyll build -s /home/taha/jekyll/chepec -d /var/www/blog`, so current working directory should not matter. Anyway, I followed your advice, put the full path in the `bibliography` option, so now the jekyll-scholar part of my `_config.yml` looks like this:

    scholar:
      style: apa
      locale: en
      sort_by: none
      order: ascending
      source: ./_bibliography
      bibliography: "/home/taha/jekyll/chepec/_bibliography/references.bib"
      bibliography_template: "{{reference}}"
      query: "@*"

And it works!! Fantastic. Allow me to take this opportunity to say that this is some really slick functionality you have achieved with jekyll-scholar. Thanks!

**Comment by inukshuk** [comment]
Ah, I haven't used the source and destination options before -- this actually explains the issue I think. I'll have to take a look at where jekyll stores the source folder path and use that when expanding scholar's source path.

**Comment by inukshuk** [comment]
Fixed in 4.3.2
